optuna/storages/cached_storage.py

```diff
@@ -280,18 +323,26 @@ def get_trial(self, trial_id: int) -> FrozenTrial:
         with self._lock:
             trial = self._get_cached_trial(trial_id)
             if trial is not None:
-                return copy.deepcopy(trial)
+                return trial
```

Comment:
Is performance degraded if we cache in this method? Could help leaving a comment.

Reply:
Since there is no method to know ``trial_id`` except for ``get_all_trials`` method and ``create_new_trial`` method, and trials are cached in those methods, the ``if trial is not None`` should be always `True` excluding some exceptional cases. Thus, adding codes that caches trials on cache miss won't affect the performance.
However, adding the code will indeed increase the consistency of implementation and reduce the confusion of readers. Thus, I'll the code, thank you.

Reply:
I remembered that it's non-trivial to get `study_id` from `trial_id`. `study_id` is required for the cache mechanism in this class and fetching the information requires a large chank of code. I'd like to keep the current implementation and not try to cache in this method.

Reply:
Ah you are right. That might be obvious from the code. A comment in the code might aid but it's definitely not a strong opinion.